src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -188,16 +188,14 @@ public bool AutoDetectScriptSuccess()
             if (!AutoDetect) return false;
 
             bool doAutoDetect = false;
-            object executionLock = new object();
-            long currentTimeStamp = DateTime.Now.Ticks;
-
-            lock (executionLock)
-            {
-                doAutoDetect = _previousTimeStampTicks == DateTime.MinValue.Ticks ?
-                    true : (currentTimeStamp - _previousTimeStampTicks > TimerPeriod ? true : false);
-
-                _previousTimeStampTicks = currentTimeStamp;
-            }
+            int currentTimeStamp = Environment.TickCount;
+
+            // Environment.TickCount will increment from zero to Int32.MaxValue for approximately 24.9 days, then jump to
+            // Int32.MinValue. When that happens, we may skip doing WinHttpDetectAutoProxyConfigUrl for one request which we
+            // suppose to do. Since _previousTimeStampTicks will get updated in that request, susequent requests in next 24.9
+            // days period will behave correctly.
+            doAutoDetect = currentTimeStamp - _previousTimeStampTicks > TimerPeriodInMilliseconds ? true : false;
+            _previousTimeStampTicks = currentTimeStamp;
```

Comment:
This is wrong.  The reason the regression disappeared, I suspect, is because doAutoDetect is only going to be true on the first request.  As long as another request is made within the next two minutes, it's going to set _previousTimeStampTicks to the current time, such that the next request made within the next two minutes will also skip detection.  This line needs to be moved to inside of the doAutoDetect if block, so that the time is only updated when we're going to refresh, at which point I suspect you'll see a regression re-emerge.

I sincerely doubt the regression disappearance has anything at all to do with the cost of Int32 vs Int64 arithmetic.